### PR TITLE
Updating to Eslint 9

### PR DESCRIPTION
**Pull request by reosarevok**
This updates our Eslint setup to Eslint 9 (we had been waiting for all the plugins we use to support it and the last one, `eslint-plugin-import`, finally got there last week).

There's a few issues, including a crash in `no-unused-vars` in some flow type pages (https://github.com/eslint/eslint/issues/18876) which might or might not get fixed on `eslint` or `hermes`. That said, for now we are fine just disabling the `no-unused-vars` rule on the three affected files.

### CODE REVIEW

File: script/dbdefs_to_js.pl [hunk 1]
```diff
@@ -147,19 +162,17 @@ sub get_value {
...
+            $browser_code .= $browser_line;
             push @all_client_defs, $def;
         }
     }
 }
 
-# Continue to allow importing as `import DBDefs from ...` by providing a
-# default export.
-$server_code .= 'export default {' . (join q(, ), @all_server_defs) . "};\n";
-$client_code .= 'export default {' . (join q(, ), @all_client_defs) . "};\n";
```
Comment (mwiencek) [inline script/dbdefs_to_js.pl:158]: Looks like these were the only real uses of `@all_server_defs` and `@all_client_defs`, so we can remove their remaining references too.

File: root/static/scripts/relationship-editor/utility/mergeDates.js [hunk 2]
```diff
@@ -26,13 +26,13 @@ export default function mergeDates(
...
     return null;
   }
-  /* eslint-disable @stylistic/js/no-multi-spaces */
+
   const mergedDate: PartialDateT = {
     day:   ad ?? bd ?? null,
     month: am ?? bm ?? null,
     year:  ay ?? by ?? null,
   };
-  /* eslint-enable @stylistic/js/no-multi-spaces */
```
Comment (mwiencek) [inline root/static/scripts/relationship-editor/utility/mergeDates.js:35]: I'm curious why this is relevant to the eslint v9 package upgrade specifically if the stylistic rules are in a separate package, and nothing from `@stylistic` was updated in yarn.lock here -- or was this meant to be in a different commit?

**Review by mwiencek — COMMENTED** [review]
Thanks very much for this, it looks great overall but might need a few small updates still.

File: root/static/scripts/guess-case/MB/GuessCase/Main.js [hunk 3]
```diff
@@ -72,56 +73,56 @@ class GuessCase {
 
     this.entities = {
       area: {
-        guess: this.guess('area', 'process'),
-        sortname: this.guess('area', 'guessSortName'),
+        guess: this.guess('area', 'process', this.regexes),
+        sortname: this.guess('area', 'guessSortName', this.regexes),
```
[the remark this answers is not in the document]
Comment (reosarevok) [reply root/static/scripts/guess-case/MB/GuessCase/Main.js:77]: Fair point - I still need to do `const regexes = this.regexes` inside `guess` to use it in the returned function, but it seems to still work (tests pass) if I do so, so let's do that.

File: root/static/scripts/relationship-editor/utility/mergeDates.js (hunk 2 above)
Comment (reosarevok) [reply root/static/scripts/relationship-editor/utility/mergeDates.js:35]: That's because of this bit of the commit message:

```
In some places, eslint let us know the rules we were disabling are no longer
erroring (probably due to now fixed eslint bugs). As such, we can remove
those disable lines.

```

Comment (mwiencek) [reply root/static/scripts/relationship-editor/utility/mergeDates.js:35]: Yeah, I read that part of the commit but I think I misunderstood what it was implying:

I had thought the `@stylistic/js/no-multi-spaces` rule was part of the `@stylistic/eslint-plugin` package rather than `eslint` package, so was wondering why upgrading the latter would cause this (apparently) unrelated rule to no longer error.

What I seem to have misunderstood is that the feature to warn about unused "disable" directives is itself a new feature, and these specific directives were *already* unused before the commit.

Comment (reosarevok) [reply root/static/scripts/relationship-editor/utility/mergeDates.js:35]: Yeah, that's what I was trying to say, maybe not very clearly :sweat_smile:

Comment (mwiencek) [reply root/static/scripts/relationship-editor/utility/mergeDates.js:35]: You could say something like, "as a new feature, ESLint now lets us know where rules we are disabling are no longer warning or erroring."

Comment (reosarevok) [reply root/static/scripts/relationship-editor/utility/mergeDates.js:35]: I reworded the commit to say something to that effect :)

**Comment by mwiencek** [comment]
@brainzbot test this please

**Comment by mwiencek** [comment]
@brainzbot test this please

**Review by mwiencek — APPROVED** [review]
Merge once Selenium tests pass!
